packages/frontend-app-api/src/extensions/AppRoot.tsx

```diff
@@ -40,26 +47,43 @@ export const AppRoot = createExtension({
   attachTo: { id: 'app', input: 'root' },
   inputs: {
     signInPage: createExtensionInput(
-      {
-        component: createSignInPageExtension.componentDataRef,
-      },
+      { component: createSignInPageExtension.componentDataRef },
       { singleton: true, optional: true },
     ),
     children: createExtensionInput(
-      {
-        element: coreExtensionData.reactElement,
-      },
+      { element: coreExtensionData.reactElement },
       { singleton: true },
     ),
+    elements: createExtensionInput(
+      { element: coreExtensionData.reactElement },
+      { optional: true },
+    ),
+    wrappers: createExtensionInput(
+      { component: createAppRootWrapperExtension.componentDataRef },
+      { optional: true },
+    ),
   },
   output: {
     element: coreExtensionData.reactElement,
   },
   factory({ inputs }) {
+    let content: React.ReactNode = (
+      <>
+        {inputs.elements.map(el => (
+          <Fragment key={el.node.spec.id}>{el.output.element}</Fragment>
+        ))}
+        {inputs.children.output.element}
```

Comment:
Realizing it may make sense to only have one of children or elements. Then again that may create some DOM ordering trouble. We're already playing with fire a bit by doing global visual elements in non-deterministic order here, so maybe best to make sure that the actual app contents is always the last DOM child here.

Reply:
Mmmyeah I don't think that'd work out. It's probably safe to document both `elements` and `wrappers` as explicitly NOT being allowed to be visual (at least not in the inline, layout-affecting sense), and `children` SHOULD be visual - making the distinction between them valuable.